### ISSUE TITLE
Memory leak in knex v0.13.0

### ISSUE THREAD

**Issue by kiralex** [tgriesser/knex#2165]
Hi, 

When importing, then destroying, then reset modules a lot of times with jest, I have some warning about memory leak `MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 SIGABRT listeners added. Use emitter.setMaxListeners() to increase limit`

I think that `destroy()` leave listener that should be removed. 

Here is the (useless) code to reproduce the issue : 

```js
//@flow

import k from 'knex';
import path from 'path';

const config = {
  client: 'sqlite3',
  connection: {
    filename: path.join(__dirname, './database.sqlite'),
  },
  pool: {
    min: 1,
    max: 1,
  },
  migrations: {
    tableName: 'knex_migrations',
  },
  useNullAsDefault: true,
  timezone: 'UTC',
};

let knex;

describe('pppppppppp', () => {
  beforeEach(async () => {
    jest.resetModules();
    knex = k(config);
  });

  afterEach(async () => {
    await knex.destroy();
  });

  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
  test('test', () => {});
});
```
the output : 
<a href="https://asciinema.org/a/Ytrq2GPKofh0R9vmmL7A5cYEl?t=11" target="_blank"><img src="https://asciinema.org/a/Ytrq2GPKofh0R9vmmL7A5cYEl.png" /></a>

Is it a bad use or a bug ? 
Thanks in advance.

**Comment by kiralex** [comment]
it seems to be the same issue as #2106

**Comment by elhigu** [comment]
Thanks for report, would be great to add this kind of test (and fixes ofc) to test suite.

**Comment by wubzz** [comment]
Can be reproduced as long as `jest.resetModules()` is called. I tested this example against branch of #2128 and there the warning no longer exists. So presumably that PR fixes the memory leak by upgrading `generic-pool` version.

**Comment by fl0w** [comment]
I've experienced this when using jest specifically myself. I'm wondering if this simply isn't a knex issue, but rather a jest issue. I do not experience equivalent issues in a project using mocha (completely different projects, the setup is quite similar, supertest/koa2/objection.

**Comment by fl0w** [comment]
@kiralex if you explicitly call `jest.clearAllTimers()` in `afterEach`, does it fix it for you?

**Comment by kiralex** [comment]
I will try. ☺️ I will give you my feedback after.

Le 27 juil. 2017 23:50, "Martin Iwanowski" <notifications@github.com> a
écrit :

**Comment by kiralex** [comment]
@fl0w nop, it does not fix the issue

**Comment by elhigu** [comment]
@kiralex  Does it leak when running initialization and destroy sequence without jest?

**Comment by wubzz** [comment]
From my previous tests this should be fixed by #2208, so closing this for now. If it's still an issue in upcoming 0.14, feel free to reopen.
